site24x7/oauth/oauth.go

```diff
@@ -0,0 +1,187 @@
+package oauth
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+)
+
+const (
+	tokenDomain = "https://accounts.zoho.com"
+	tokenApi    = "/oauth/v2/token"
+	timeout     = time.Second
+)
+
+func setRequestHeaders(request *http.Request, accessToken string) {
+	request.Header.Set("Content-Type", "application/json;charset=UTF-8")
+	request.Header.Set("Accept", "application/json; version=2.0")
+	request.Header.Set("Authorization", "Zoho-oauthtoken " + accessToken)
+}
+
+func getUrl(urlValues url.Values) string {
+	baseAccurl := fmt.Sprintf("%s%s", tokenDomain, tokenApi)
+	urlToReturn := fmt.Sprintf("%s?%s", baseAccurl, urlValues.Encode())
+	return urlToReturn
+}
+
+type Authenticator struct {
+	// Protects tkns.AccessToken (updated in refresh goroutine)
+	sync.Mutex
+	tkns      tokens
+	storePath string
+}
+
+func (ator *Authenticator) getClient() *http.Client {
+	trTls11 := &http.Transport{
```

Comment:
Should it be `trTls13` (actually uses `tls.VersionTLS13`)?

nit:

 ```suggestion
	trTLS11 := &http.Transport{
```

Reply:
no clue, i got this stuff from the sample code